src/main/java/com/minecolonies/core/client/gui/townhall/WindowTownHallDeleteAbandonColony.java

```diff
@@ -0,0 +1,119 @@
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+
+import static com.minecolonies.api.util.constant.Constants.MOD_ID;
+import static com.minecolonies.api.util.constant.WindowConstants.*;
+
+/**
+ * UI to delete or abandon an old colony.
+ */
+public class WindowTownHallDeleteAbandonColony extends AbstractWindowSkeleton
+{
+    /**
+     * String constants.
+     */
+    private static final String DELETE_PROCEED = "com.minecolonies.core.gui.colony.delete.proceed";
+    private static final String ABANDON_PROCEED = "com.minecolonies.core.gui.colony.abandon.proceed";
+    private static final String DELETE_WARNING =    "com.minecolonies.core.gui.colony.delete.warning";
+    private static final String ABANDON_WARNING =    "com.minecolonies.core.gui.colony.abandon.warning";
+
+    /**
+     * Townhall position
+     */
+    private final BlockPos pos;
+
+    public WindowTownHallDeleteAbandonColony(final Player player, final BlockPos pos, final Level world, final String oldColonyName, final BlockPos oldColonyPos, final int oldColonyId)
+    {
+        super(MOD_ID + TOWNHALL_DELETE_ABANDON_GUI);
+        this.pos = pos;
+
+        registerButton(BUTTON_CANCEL, this::close);
+        registerButton(BUTTON_PICKUP_BUILDING, this::pickup);
+        registerButton(BUTTON_DELETE, this::deleteColony);
+        registerButton(BUTTON_ABANDON, this::abandonColony);
+        registerButton("cancelaction", this::cancel);
+
+        registerButton(BUTTON_CONFIRM_DELETE, this::confirmDeleteColony);
+        registerButton(BUTTON_CONFIRM_ABANDON, this::confirmAbandonColony);
+
+        if (MineColonies.getConfig().getServer().allowInfiniteColonies.get())
+        {
+            this.findPaneOfTypeByID("abandon", ButtonImage.class).show();
+            this.findPaneOfTypeByID("warningtext", Text.class).setText(Component.translatable(ABANDON_WARNING, oldColonyPos.getX(), oldColonyPos.getY(), oldColonyPos.getZ(), Component.literal(oldColonyName).withStyle(ChatFormatting.DARK_RED)));
+        }
+        else
+        {
+            this.findPaneOfTypeByID("abandon", ButtonImage.class).hide();
+            this.findPaneOfTypeByID("warningtext", Text.class).setText(Component.translatable(DELETE_WARNING, oldColonyPos.getX(), oldColonyPos.getY(), oldColonyPos.getZ(), Component.literal(oldColonyName).withStyle(ChatFormatting.DARK_RED)));
+        }
+    }
+
+    private void confirmAbandonColony()
+    {
+        Network.getNetwork().sendToServer(new ColonyAbandonOwnMessage());
+        Network.getNetwork().sendToServer(new GetColonyInfoMessage(pos));
+        close();
+    }
+
+    private void confirmDeleteColony()
+    {
+        Network.getNetwork().sendToServer(new ColonyDeleteOwnMessage());
+        Network.getNetwork().sendToServer(new GetColonyInfoMessage(pos));
+        close();
+    }
+
+    private void abandonColony()
+    {
+        this.findPaneOfTypeByID("dialog", Image.class).show();
```

Comment:
these methods are code smell imho, use view to wrap disabling/enabling of element groups, or use switch view (if applicable)

Reply:
might seem like it, but depending on the case I enable/disable the abandoned one separately anyway

Reply:
honestly, could be one cancel button but whatever